minijanus.js

```diff
@@ -88,9 +88,19 @@ JanusSession.prototype.create = function() {
 
 /** Destroys this session. **/
 JanusSession.prototype.destroy = function() {
-  // Note that this message, unlike others, does *not* receive a response from janus, so there is no promise returned.
-  this.send("destroy");
+  // This message will not receive a response from the server.
+  // It will also be rejected at the bottom of this method so we add an empty catch handler.
+  this.send("destroy").catch((e) => {});
+
   this._killKeepalive();
+
+  for (var txnId in this.txns) {
+    if (this.txns.hasOwnProperty(txnId)) {
+      var txn = this.txns[txnId];
+      clearTimeout(txn.timeout);
```

Comment:
I think you need to check if `txn.timeout` is `null` in case timeouts are disabled. See the corresponding code in `JanusSession.receive`.

Reply:
`clearTimeout(null)` and `clearTimeout(undefined)` do not error so I usually do not add this check. The standard doesn't actually say anything about this so perhaps we should do it? I think this is one of those habits I've gained over the years and never really thought about.

https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#dom-cleartimeout